### PR TITLE
#1239 Input path validation

### LINKED ISSUE #1239: Check if STD path exists before running conformance
## Background
Currently, if standardization folder does not exist during conformance, the error is NoFileFound and a message that metrics failed.

## Feature
Before anything from conformance, check if standardization folder exists and if not, throw an error.

## Proposed Solution [Optional]
```scala
if (!fsUtils.hdfsExists(pathCfg.stdPath)) {
      throw new IllegalStateException(
        s"Path ${pathCfg.stdPath} does not exist. Conformance has no input data")
    }
```

### PR DESCRIPTION
**Pull request by AdrianOlosutean**
Closes #1239

### CODE REVIEW

**Review by dk1844 — APPROVED** [review]

**Review by benedeki — DISMISSED** [review]

**Comment by benedeki** [comment]
- [x] code reviewed
- [ ] pulled
- [ ] built
- [ ] run

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala [hunk 1]
```diff
@@ -93,10 +93,11 @@ trait StandardizationExecution extends CommonJobExecution {
     }
   }
 
-  override def validateOutputPath(fsUtils: FileSystemVersionUtils, pathConfig: PathConfig): Unit = {
+  override def validatePaths(fsUtils: FileSystemVersionUtils, pathConfig: PathConfig): Unit = {
     log.info(s"raw path: ${pathConfig.rawPath}")
     log.info(s"standardization path: ${pathConfig.standardizationPath}")
-    validateIfPathAlreadyExists(fsUtils: FileSystemVersionUtils, pathConfig.standardizationPath)
+    validateInputPath(fsUtils, pathConfig.rawPath)
+    validateIfOutputPathAlreadyExists(fsUtils: FileSystemVersionUtils, pathConfig.standardizationPath)
```
Comment (Zejnilovic) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala:100]: why is there a type `FileSystemVersionUtils` declaration?

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala [hunk 2]
```diff
@@ -100,16 +100,24 @@ trait CommonJobExecution extends ProjectMetadata {
...
+  protected def validateIfOutputPathAlreadyExists(fsUtils: FileSystemVersionUtils, path: String): Unit = {
     if (fsUtils.hdfsExists(path)) {
       throw new IllegalStateException(
         s"Path $path already exists. Increment the run version, or delete $path"
       )
     }
   }
 
+  protected def validateInputPath(fsUtils: FileSystemVersionUtils, path: String): Unit = {
+    if (!fsUtils.hdfsExists(path)) {
```
Comment (Zejnilovic) [inline spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:114]: Question: Will this hold in case of cloud? @dk1844

**Review by Zejnilovic — COMMENTED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala (hunk 2 above)
Comment (benedeki) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:114]: Actually won't it prevent to run Enceladus from local path as well? If yes, is it something we want?

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala (hunk 1 above)
Comment (benedeki) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala:100]: Good catch. I guess a paste leftover?

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala (hunk 2 above)
Comment (dk1844) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:114]: Short answer: it's sort of ok as-is.

Long answer: Cosidering the facts that this is on `develop` branch that has no notion of features in `aws-poc` and that the method name has changed slightly in the `aws-poc` (`fsUtils.exists(path)`), it will still check the (non-)existence of the path. It is important to note (and that also touches @benedeki's question), that in the cloud-version, at least, currently, the `fsUtils` instance is tied to the `FileSystem` it was created with (hdfs-bound, s3-bound) and it will check the path at target. So it's a _what-you-pass-is-what-you-get_ situation.

Bottom line, (even though I understand the motivation) it is a bit weird to ask these questions here for the reasons outlined above.

File: spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala (hunk 1 above)
Comment (AdrianOlosutean) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala:100]: Yes

File: spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala (hunk 2 above)
Comment (benedeki) [reply spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala:114]: OK, not relevant for this PR. But can we get to the situation that read and write file systems are of different type?

**Review by benedeki — APPROVED** [review]
